runtime/lua/vim/lsp/sync.lua

```diff
@@ -202,14 +205,14 @@ end
 --- prev_end_range is the text range sent to the server representing the changed region.
 --- curr_end_range is the text that should be collected and sent to the server.
 --
----@param prev_lines table list of lines
----@param curr_lines table list of lines
----@param start_range table
+---@param prev_lines string[] list of lines
+---@param curr_lines string[] list of lines
+---@param start_range vim.lsp.sync.Range
+---@param firstline integer
 ---@param lastline integer
 ---@param new_lastline integer
 ---@param offset_encoding string
----@return integer|table end_line_idx and end_col_idx of range
----@return table|nil end_col_idx of range
+---@return vim.lsp.sync.Range, vim.lsp.sync.Range
```

Comment:
this function is quite long, please consider adding a quick hint about what this would return.

Reply:
Ehhh unfortunately I don't understand what's going on well enough to write a helpful comment here...